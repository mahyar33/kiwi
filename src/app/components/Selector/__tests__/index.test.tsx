import React from 'react';
import { prettyDOM, render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Selector } from '../index';

const renderWithProps = (props: Parameters<typeof Selector>[number]) =>
  render(<Selector {...props} />);

describe('<Selector />', () => {
  const props: Parameters<typeof Selector>[number] = {
    placeholder: 'placeholder',
    name: 'Selector',
    options: [
      { text: 'yes', value: 'yes', id: 0 },
      { text: 'no', value: 'no', id: 1 },
    ],
    onChange: jest.fn(),
    onSelectItem: jest.fn(),
  };
  it('should render an <input> tag', () => {
    const selector = renderWithProps(props);
    expect(selector.container.querySelector('input')).toBeInTheDocument();
  });
  it('should render an loading indicator', () => {
    props.loading = true;
    props.expanded = true;
    const selector = renderWithProps(props);
    expect(selector.container.querySelector('svg')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const selector = renderWithProps(props);
    expect(selector.container.firstChild).toMatchSnapshot();
  });
  it('should have props', () => {
    const selector = renderWithProps(props);
    expect(selector.container.querySelector('input')).toHaveAttribute(
      'placeholder',
      props.placeholder,
    );
  });
  it('should call onSelectItem', () => {
    props.loading = false;
    const selector = renderWithProps(props);
    userEvent.click(selector.getByText('yes', { exact: false }));
    expect(props.onSelectItem).toHaveBeenCalledTimes(1);
  });
  it('should call onChange', () => {
    const selector = renderWithProps(props);
    userEvent.type(selector.container.querySelector('input')!, 'H');
    expect(props.onChange).toHaveBeenCalledTimes(1);
  });
  it('should have 2 options', () => {
    const selector = renderWithProps(props);
    expect(
      selector.container.querySelectorAll('div > div > div>div ').length,
    ).toBe(2);
  });
});
