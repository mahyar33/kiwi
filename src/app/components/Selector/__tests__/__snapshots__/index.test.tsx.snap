// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Selector /> should match snapshot 1`] = `
.c0 {
  position: relative;
}

.c2 {
  position: absolute;
  width: 100%;
  text-align: center;
  background-color: #fff;
  border: 1px solid #ccc;
  border-top: 0;
  box-shadow: inset 0px 0px 3px 0px #6d6c6c8a;
  max-height: 110px;
  overflow: unset;
}

.c2 svg {
  width: 30px;
}

.c1 {
  display: block;
  width: 100%;
  height: 2.5rem;
  padding: 6px 12px;
  line-height: 1.5;
  color: black;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ccc;
}

.c1:focus-visible {
  outline: unset;
}

.c3 {
  -webkit-animation: cqkDSr 2.625s linear infinite;
  animation: cqkDSr 2.625s linear infinite;
  height: 3rem;
  width: 3rem;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c4 {
  -webkit-animation: caVMmK 1.5s ease-in-out infinite;
  animation: caVMmK 1.5s ease-in-out infinite;
  stroke: rgba(215,113,88,1);
  stroke-linecap: round;
}

<div
  class="c0"
>
  <input
    class="c1"
    placeholder="placeholder"
    type="search"
    value=""
  />
  <div
    class="c2"
  >
    <svg
      class="c3"
      viewBox="-24 -24 48 48"
    >
      <circle
        class="c4"
        cx="0"
        cy="0"
        fill="none"
        r="20"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;
